src/becca/similarity.ts

```diff
@@ -237,23 +242,23 @@ async function findSimilarNotes(noteId) {
         return [];
     }
 
-    let dateLimits;
+    let dateLimits: DateLimits;
 
     try {
         dateLimits = buildDateLimits(baseNote);
     }
-    catch (e) {
+    catch (e: any) {
         throw new Error(`Date limits failed with ${e.message}, entity: ${JSON.stringify(baseNote.getPojo())}`);
     }
 
     const rewardMap = buildRewardMap(baseNote);
-    let ancestorRewardCache = {};
+    let ancestorRewardCache: Record<string, number> = {};
     const ancestorNoteIds = new Set(baseNote.getAncestors().map(note => note.noteId));
     ancestorNoteIds.add(baseNote.noteId);
 
     let displayRewards = false;
 
-    function gatherRewards(text, factor = 1) {
+    function gatherRewards(text?: string | null, factor: number = 1) {
```

Comment:
Same as previous:

> I think the original intent of the code was for text to be optional. If I remove the `undefined | null` then it can crash if `branch.prefix` is null or `note.mime` is null.
> 
> I think the goal of similarity index is to add as many relevant text as possible, where there is no such value we simply ignore it. This avoids having to add manual null checks for four lines of code.